src/odemis/driver/smarpod.py

```diff
@@ -350,29 +349,34 @@ def __init__(self, name, role, locator, axes=None, **kwargs):
         # will take care of executing axis move asynchronously
         self._executor = CancellableThreadPoolExecutor(1)  # one task at a time
 
-        # Check referencing, and reference if not referenced
+        axes_ref = {a: False for a, i in self.axes.items()}
+        # VA dict str(axis) -> bool
         self.referenced = model.VigilantAttribute(axes_ref, readonly=True)
         referenced = c_int()
         self.core.Smarpod_IsReferenced(self._id, byref(referenced))
         if not referenced.value:
-            logging.debug("SmartPod is not referenced. The device will not function until referencing occurs.")
+            logging.debug("SmarPod is not referenced. The device will not function until referencing occurs.")
+            if ref_on_init:
+                self.reference().result()
         else:
-            logging.debug("SmartPod is referenced")
+            logging.debug("SmarPod is referenced")
 
+        # Use a default actuator speed
+        self.SetSpeed(actuator_speed)
         self._speed = self.GetSpeed()
         self._accel = self.GetAcceleration()
         try:
             self._updatePosition()    # will fail if not referenced.
-        except SmartPodError as ex:
-            if ex.errno == SmartPodDLL.SMARPOD_NOT_REFERENCED_ERROR:
-                logging.info("Must reference SmartPod controller before use.")
+        except SmarPodError as ex:
+            if ex.errno == SmarPodDLL.SMARPOD_NOT_REFERENCED_ERROR:
+                logging.info("Must reference SmarPod controller before use.")
```

Comment:
That's the second log about the same thing. Just indicate that the position is unknown until the device is referenced.
Please also make sure that in this case, `.position` contains still a dict, with all the axes. I'd suggest to just put `0` as the position for each axis (as it'll probably be the position once it's referenced).